docs/admin/clustering/scale/auto.md

```diff
@@ -0,0 +1,227 @@
+echo ${numnodes}
+```
+Output
+```json
+3
+```
+So, if you want to scale out, you can call the scale API. Note that the `clusterid` is needed.
+
+```bash
+curl -u $auth_data -X PUT "https://console.cratedb.cloud/api/v2/clusters/${clusterid}/scale/" -H "accept: application/json" -H "Content-Type: application/json" -d "{\"product_unit\":2}"
+```
+
+Extra info:
+```product_unit``` is the number of nodes minus 1, as 1 is considered the minimum number. In case you want to increase a 3 node cluster you need to use 3 as the value for product_unit.
+
+```bash
+jobid=$(curl -u $auth_data -X PUT "https://console.cratedb.cloud/api/v2/clusters/${clusterid}/scale/" -H "accept: application/json" -H "Content-Type: application/json" -d "{\"product_unit\":$numnodes}" | jq -r '.last_async_operation.id')
+```
+The job can be used to query the operations API to get the info on the status.
+
+```bash
+# check status:
+curl -s -u $auth_data https://console.cratedb.cloud/api/v2/clusters/${clusterid}/operations/ | jq --arg jobid "$jobid" '.operations[] | select(.id == $jobid)'
+```
+The `status` should change from `IN_PROGRESS` to `SUCCEEDED`:
+```json
+{
+  "dc": {
+    "created": "2024-02-23T13:04:22.960000+00:00",
+    "modified": "2024-02-23T13:06:52.157000+00:00"
+  },
+  "entity": "CLUSTER",
+  "entity_id": "xxxxx-b44e-446f-adca-a092fd1df0aa",
+  "feedback_data": {
+    "message": "Successfully scaled cluster to 4 nodes."
+  },
+  "id": "xxxxx-3d33-4c4a-8c9c-93202931ed1c",
+  "non_sensitive_data": {
+    "current_num_nodes": 3,
+    "current_product_unit": 2,
+    "target_num_nodes": 4,
+    "target_product_unit": 3
+  },
+  "status": "SUCCEEDED",
+  "type": "SCALE"
+}
+```
+
+
+## Automatic Scaling
+
+We created a Python script that can be used as a starting point to enable autoscaling (out and back). In this script, we check the number of shards in a cluster, and when this crosses a threshold, the cluster will be scaled out. When the threshold goes a certain number, the cluster will be scaled back.
+
+Download the [autoscaling script from GitHub] and run it as indicated in the README.
+
+You can control the autoscale behavior by defining the ``MAX_NUM_SHARDS`` and by tweaking the number of shards in the cluster. For example, by creating an extra table with x number of shards to see the autoscale in action. In the script, the ``MAX_NUM_SHARDS`` is set to 30, but this can be changed to a number of your liking. It is also good to understand that the script will trigger a scale-out when the number of shards is higher than 80% of the ``MAX_NUM_SHARDS``. Assuming the default number of 30, this will mean that when the number of shards exceeds 24, the autoscale will kick in.
+
+In this example, I created a 3-node CrateDB Cloud cluster and created this table:
+
+``` sql
+CREATE TABLE ta (
+    "keyword" TEXT INDEX using fulltext
+    ,"ts" TIMESTAMP
+    ,"day" TIMESTAMP GENERATED ALWAYS AS date_trunc('day', ts)
+    )
```

Comment:
```suggestion
    "keyword" TEXT INDEX using fulltext,
    "ts" TIMESTAMP,
    "day" TIMESTAMP GENERATED ALWAYS AS date_trunc('day', ts)
)
```
Don't want to start a formatting fight, but this style of "comma at the beginning" isn't used anywhere else in the documentation to my knowledge, so let's keep it consistent with the predominant style.

Reply:
Thanks. I just copied the content verbatim from the community forum. I wanted to spend a polishing iteration on it, but most probably I would not have spotted this. I will also take it into consideration for future refactorings.